### PR TITLE
Fix: adjust WalletConnect events

### LINKED ISSUE #2720: Safe-as-an-owner+WC: add safe{wallet} app icon
## Environment
 - Browser: Chrome/FF
 - Wallet: MetaMask/safe-as-an-owner
 - Chain: any

## Steps to reproduce
 1. Connect safe-as-an-owner via WC to parent safe
Current result: Safe{Wallet} web app doesn't have logo

## Expected result
Add safe{Wallet} web app logo 
## Screenshots
image

image

### PR DESCRIPTION
**Pull request by katspaugh**
## What it solves

* Change the "WC show hints" event to trigger on i-icon click + add a "WC hide hints" event - verified
* Add an event for all RPC requests made through WalletConnect – "WC request"
* Don't ignore chain switch errors
* Display errors in two lines: summary and details - changed , the error is displayed in a few lines. The connection code is too long for one line
* Limit the displayed dapp name to 22 characters in the session list - verified
* Reduce the time to show a connect/disconnect message to 1 second (was 2) - verified
* Fix Safe logo for WalletConnect (resolves #2720) - verified
* Show the new WC popup when the old WC Safe App is opened - verified

### CODE REVIEW

File: src/components/walletconnect/WcErrorMessage/index.tsx [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
 const WcErrorMessage = ({ error, onClose }: { error: Error; onClose: () => void }) => {
+  const message = error.message || 'An error occurred'
+  const [summary, details] = splitError(message)
+
   return (
     <div className={css.errorContainer}>
       <WcLogoHeader error />
 
       <Typography title={error.message} className={css.errorMessage} mb={3}>
-        {error.message || 'An error occurred'}
```
Comment (usame-algan) [inline src/components/walletconnect/WcErrorMessage/index.tsx:11]: The error message layout looks a bit stretched out. Makes it seem like the details belong to the button. I think we have an `ErrorMessage` component. Is it possible to reuse it here?

Screenshot 2023-11-01 at 09 34 44

Comment (katspaugh) [reply src/components/walletconnect/WcErrorMessage/index.tsx:11]: Thank you, fixed!

Screenshot 2023-11-01 at 10 29 33

**Review by usame-algan — APPROVED** [review]

**Comment by liliya-soroka** [comment]
Error view :
image

**Comment by liliya-soroka** [comment]
@katspaugh , could you check why the tx doesn't appear when the safe connected to the ENS app ( https://app.ens.domains/) on Goerli? Works on prod

**Comment by katspaugh** [comment]
Depends what action you're performing and how much gas it costs.
I'm getting "unpredictable gas cost" for extensions for example, but it works for registering a new domain.

Screenshot 2023-11-01 at 12 14 39

**Comment by liliya-soroka** [comment]
off-chain signatures from the dapp connected via WC don't have walletconnect name 
image

**Comment by katspaugh** [comment]
@liliya-soroka fixed ✅ 
Screenshot 2023-11-01 at 13 31 52

**Comment by liliya-soroka** [comment]
The message was unexpected:
1. I had 2 connections
2. tried to connect to opensea when Abitrum safe was opened in safe web UI
Current result: the message is confusing. Not sure is it about wrong chain in the safe app or on the opensea
image

**Comment by katspaugh** [comment]
I've made the message clearer.
Screenshot 2023-11-01 at 15 26 49

**Comment by liliya-soroka** [comment]
Verified.
WE are ready to create a RC

**Pull request merged**